client/src/utils/election.ts

```diff
@@ -16,6 +22,54 @@ export const getContests = ({
       ballotStyle.partyId === c.partyId
   )
 
+export const getEitherNeitherContests = (
+  contests: Contests
+): MsEitherNeitherContest[] =>
+  contests
+    .filter((c) => c.type === 'ms-either-neither')
+    .map((c) => c as MsEitherNeitherContest)
+
+export const expandEitherNeitherContests = (contests: Contests): Contests => {
+  const resultContests = [...contests]
+
+  const eitherNeitherContests = getEitherNeitherContests(contests)
+
+  eitherNeitherContests.forEach((c) => {
+    const eitherNeitherIndex = resultContests.findIndex((rc) => rc === c)
+
+    const eitherNeitherContest = {
+      id: c.eitherNeitherContestId,
+      type: 'yesno',
+      title: `${c.title} -- Either/Neither`,
+      districtId: c.districtId,
+      section: c.section,
+      description: c.description,
+      yesOption: c.eitherOption,
+      noOption: c.neitherOption,
+    } as YesNoContest
+
+    const pickOneContest = {
+      id: c.pickOneContestId,
+      type: 'yesno',
+      title: `${c.title} -- Pick One`,
+      districtId: c.districtId,
+      section: c.section,
+      description: c.description,
+      yesOption: c.firstOption,
+      noOption: c.secondOption,
+    } as YesNoContest
+
+    resultContests.splice(
+      eitherNeitherIndex,
+      1,
+      eitherNeitherContest,
+      pickOneContest
+    )
+  })
+
+  return resultContests
+}
```

Comment:
```suggestion
export const expandEitherNeitherContests = (
  contests: Contests
): Exclude<AnyContest, MsEitherNeitherContest>[] =>  contests.flatMap((contest) =>
    contest.type !== 'ms-either-neither'
      ? [contest]
      : [
          {
            type: 'yesno',
            id: contest.eitherNeitherContestId,
            title: `${contest.title} – Either/Neither`,
            districtId: contest.districtId,
            section: contest.section,
            description: contest.description,
            yesOption: contest.eitherOption,
            noOption: contest.neitherOption,
          },
          {
            type: 'yesno',
            id: contest.pickOneContestId,
            title: `${contest.title} – Pick One`,
            districtId: contest.districtId,
            section: contest.section,
            description: contest.description,
            yesOption: contest.firstOption,
            noOption: contest.secondOption,
          },
        ]
  )
```

Functionally the same, except I replaced `--` with `–`. Also made the return type more specific.